nilearn/regions/tests/test_signal_extraction.py

```diff
@@ -9,490 +9,492 @@
+    with pytest.raises(DimensionError, match=_3D_EXPECTED_ERROR_MSG):
+        signals_to_img_labels(
+            signals=img_4D, labels_img=img_3D, mask_img=wrong_dim_mask_img
+        )
 
-    labels_img = nibabel.Nifti1Image(labels_data, affine)
+    with pytest.raises(DimensionError, match=_3D_EXPECTED_ERROR_MSG):
+        img_to_signals_maps(
+            imgs=img_4D, maps_img=img_4D, mask_img=wrong_dim_mask_img
+        )
 
-    labels_4d_data = np.zeros((shape) + (2,))
-    labels_4d_data[..., 0] = labels_data
-    labels_4d_data[..., 1] = labels_data
-    labels_4d_img = nibabel.Nifti1Image(labels_4d_data, np.eye(4))
 
-    # Without mask
-    # from labels
-    data_img = signal_extraction.signals_to_img_labels(signals, labels_img)
-    data = get_data(data_img)
-    assert data_img.shape == (shape + (n_instants,))
-    assert np.all(data.std(axis=-1) > 0)
+"""errors img_to_signals_labels"""
+
+
+@pytest.mark.parametrize(
+    "shape, affine, error_msg",
+    [
+        ((8, 9, 11), AFFINE, _TEST_SHAPE_ERROR_MSG),
+        (SHAPE, 2 * AFFINE, _TEST_AFFINE_ERROR_MSG),
+    ],
+)
+def test_img_to_signals_labels_bad_inputs(
+    img_4D, img_3D, shape, affine, error_msg
+):
+    bad_labels_img = nibabel.Nifti1Image(np.zeros(shape), affine)
 
-    # verify that 4D label images are refused
-    with pytest.raises(DimensionError, match=_TEST_DIM_ERROR_MSG):
-        signal_extraction.img_to_signals_labels(data_img, labels_4d_img)
+    with pytest.raises(ValueError, match=error_msg):
+        img_to_signals_labels(imgs=img_4D, labels_img=bad_labels_img)
 
+    with pytest.raises(ValueError, match=error_msg):
+        img_to_signals_labels(
+            imgs=img_4D, labels_img=bad_labels_img, mask_img=img_3D
+        )
+
+    bad_mask_img = nibabel.Nifti1Image(np.zeros(shape), affine)
+
+    with pytest.raises(ValueError, match=error_msg):
+        img_to_signals_labels(
+            imgs=img_4D, labels_img=img_3D, mask_img=bad_mask_img
+        )
+
+
+def test_img_to_signals_labels_error_strategy(img_4D, img_3D):
+    with pytest.raises(ValueError, match="Invalid strategy"):
+        img_to_signals_labels(imgs=img_4D, labels_img=img_3D, strategy="foo")
+
+
+"""errors signals_to_img_labels"""
+
+
+@pytest.mark.parametrize(
+    "shape, affine, error_msg",
+    [
+        ((8, 9, 11), AFFINE, _TEST_SHAPE_ERROR_MSG),
+        (SHAPE, 2 * AFFINE, _TEST_AFFINE_ERROR_MSG),
+    ],
+)
+def test_signals_to_img_labels_bad_inputs(
+    img_4D, img_3D, shape, affine, error_msg
+):
+    bad_labels_img = nibabel.Nifti1Image(np.zeros(shape), affine)
+
+    with pytest.raises(ValueError, match=error_msg):
+        signals_to_img_labels(
+            signals=img_4D, labels_img=bad_labels_img, mask_img=img_3D
+        )
+
+    bad_mask_img = nibabel.Nifti1Image(np.zeros(shape), affine)
+
+    with pytest.raises(ValueError, match=error_msg):
+        signals_to_img_labels(
+            signals=img_4D, labels_img=img_3D, mask_img=bad_mask_img
+        )
```

Comment:
I think we can split here also:

```python
def test_signals_to_img_labels_bad_label_img_input(...):
    bad_img = nibabel.Nifti1Image(np.zeros(shape), affine)
    with pytest.raises(ValueError, match=error_msg):
        signals_to_img_labels(signals=img_4D, labels_img=bad_img, mask_img=img_3D)

def test_signals_to_img_labels_bad_label_input(...):
    bad_img = nibabel.Nifti1Image(np.zeros(shape), affine)
    with pytest.raises(ValueError, match=error_msg):    
        signals_to_img_labels(signals=img_4D, labels_img=img_3D, mask_img=bad_img)
```

Reply:
I meant `bad_mask_input` for the second test, sorry.

Reply:
done